controller/controller.go

```diff
@@ -107,14 +114,15 @@ func NewController(f ...ControllerOption) *Controller {
 	}
 
 	controller := &Controller{
-		name:      opts.name,
-		logger:    opts.logger,
-		client:    opts.client,
-		manager:   opts.manager,
-		cache:     &watchableCacheStore{},
-		topology:  newGatewayAPITopologyBuilder(opts.policyKinds, opts.objectKinds, opts.objectLinks),
-		runnables: map[string]Runnable{},
-		reconcile: opts.reconcile,
+		name:               opts.name,
+		logger:             opts.logger,
+		client:             opts.client,
+		manager:            opts.manager,
+		cache:              &watchableCacheStore{},
+		topology:           newGatewayAPITopologyBuilder(opts.policyKinds, opts.objectKinds, opts.objectLinks),
+		runnables:          map[string]Runnable{},
+		reconcile:          opts.reconcile,
+		allowTopologyLoops: opts.allowTopologyLoops,
```

Comment:
What about making this a field of the [`gatewayAPITopologyBuilder`](https://github.com/Kuadrant/policy-machinery/blob/85837a3f6a5ae4b84594de2124c5338f678a57f3/controller/topology_builder.go#L20) struct? The `propagate` func of the controller would not raise the error, but the topology builder's `Build` func instead. There would be no need of modifying the signature of the `Build` func.

Reply:
I think I get what your saying. Update has being pushed.